Wikitext - [WIP]

ayushi-3536 opened the pull request.

ayushi-3536 committed:
- added language model benchmark from MO-ASHA paper
- added dependencies, benchmark
- added token generation and model training code from moasha in dependencies

ayushi-3536 committed:
- added language model benchmark from MO-ASHA paper
- added dependencies, benchmark
- added token generation and model training code from moasha in dependencies

ayushi-3536 committed:
- added evaluation time as one of the objectives
- returning prediction time for evaluation time
- changed perplexity --> log_perplexity for the objective (MO-ASHA uses log perplexity)
 changed error --> accuracy
- added tqdm

ayushi-3536 committed:
- added evaluation time as one of the objectives
- returning prediction time for evaluation time
- changed perplexity --> log_perplexity for the objective (MO-ASHA uses log perplexity)
 changed error --> accuracy
- added tqdm

ayushi-3536 committed:
- func name correction

ayushi-3536 committed:
- func name correction

ayushi-3536 committed:
- load and  save tokenized file

ayushi-3536 committed:
- load and  save tokenized file

ayushi-3536 committed:
-code formatting

ayushi-3536 committed:
-code formatting

ayushi-3536 committed:
-make deterministic
-report train and eval time separately in objective func
-code formatting
-added test file

ayushi-3536 committed:
-make deterministic
-report train and eval time separately in objective func
-code formatting
-added test file

ayushi-3536 committed:
-added lock for download data
- added recipe and container file

ayushi-3536 committed:
-added lock for download data
- added recipe and container file

ayushi-3536 committed:
-make emsize sampling log based: To be discussed with team - positional encoding doesn't work for odd number, therefor log seems like perfect solution
-removed logs

ayushi-3536 committed:
-make emsize sampling log based: To be discussed with team - positional encoding doesn't work for odd number, therefor log seems like perfect solution
-removed logs

PhMueller committed:
Update Github Actions (#151)

* Update Github Actions Workflow  and drop support for singularity < 3.7

PhMueller commented on hpobench/benchmarks/mo/lm_benchmark.py:
```diff
@@ -0,0 +1,368 @@
+class LanguageModelBenchmark(AbstractMultiObjectiveBenchmark):
+
+    def __init__(self, rng: Union[np.random.RandomState, int, None] = None, **kwargs):
+        super(LanguageModelBenchmark, self).__init__(rng=rng)
```
Can you please add a short description?

PhMueller replied:
Can you please add a short description?

PhMueller commented on hpobench/benchmarks/mo/lm_benchmark.py:
```diff
@@ -0,0 +1,368 @@
+                         "nhead": 2,
+                         "ntoken": self.ntokens
+                         }
+        print("len of corpus dict", self.ntokens)
```
Please use logging.debug for printing

PhMueller replied:
Please use logging.debug for printing

PhMueller commented on hpobench/benchmarks/mo/lm_benchmark.py:
```diff
@@ -0,0 +1,368 @@
+    def __seed_everything(self):
+        """Helperfunction: Make the benchmark deterministic by setting the correct seeds"""
+        seed = self.rng.randint(0, 100000)
+        print("seed obtained", seed)
```
same here

PhMueller replied:
same here

PhMueller commented on hpobench/benchmarks/mo/lm_benchmark.py:
```diff
@@ -0,0 +1,368 @@
+        return cs
+
+    @staticmethod
+    def get_objective_names(self) -> List[str]:
```
Remove `self`.

PhMueller replied:
Remove `self`.

PhMueller commented on hpobench/benchmarks/mo/lm_benchmark.py:
```diff
@@ -0,0 +1,368 @@
+
+    @staticmethod
+    def get_fidelity_space(seed: Union[int, None] = None) -> CS.ConfigurationSpace:
+
```
Short docstring please. (numpy format: sorry i have mixed up `google`with `numpy`)

PhMueller replied:
Short docstring please. (numpy format: sorry i have mixed up `google`with `numpy`)

PhMueller commented on hpobench/benchmarks/mo/lm_benchmark.py:
```diff
@@ -0,0 +1,368 @@
+        """ Function that returns the model initialized based on the configuration and fidelity
+        """
+
+        if isinstance(config, CS.Configuration):
```
actually we don't need that check (`if isinstance`) here. 

The objective_function -wrapper casts configurations always to dicts.

PhMueller replied:
actually we don't need that check (`if isinstance`) here. 

The objective_function -wrapper casts configurations always to dicts.

PhMueller commented on hpobench/benchmarks/mo/lm_benchmark.py:
```diff
@@ -0,0 +1,368 @@
+            To prevent overfitting on a single seed, it is possible to pass a
+            parameter ``rng`` as 'int' or 'np.random.RandomState' to this function.
+            If this parameter is not given, the default random state is used.
+
```
Add `shuffle` here.

PhMueller replied:
Add `shuffle` here.

PhMueller commented on hpobench/benchmarks/mo/lm_benchmark.py:
```diff
@@ -0,0 +1,368 @@
+            start = time.time()
+            val_loss, val_acc = model.eval_fun(self.ntokens, criterion, val_data)
+            val_loss = np.clip(val_loss, 1e-10, 10)
+            print("val acc for last epoch", val_acc)
```
`logging.debug`
see above

PhMueller replied:
`logging.debug`
see above

PhMueller commented on hpobench/benchmarks/mo/lm_benchmark.py:
```diff
@@ -0,0 +1,368 @@
+            t.update()
+
+            if not np.isfinite(val_loss):
+                val_loss = 7
```
is there a reason for the 7? 

Could you please add a comment to it, if that is from the authors

PhMueller replied:
is there a reason for the 7? 

Could you please add a comment to it, if that is from the authors

PhMueller commented on hpobench/benchmarks/mo/lm_benchmark.py:
```diff
@@ -0,0 +1,368 @@
+        perplexity = math.exp(best_val_loss)
+        log_perplexity = best_val_loss
+        neg_log_perplexity = 10 - best_val_loss
+        elapsed_time = float(ts_start - time.time())
```
elapsed_time is already float

PhMueller replied:
elapsed_time is already float

PhMueller commented on hpobench/benchmarks/mo/lm_benchmark.py:
```diff
@@ -0,0 +1,368 @@
+            To prevent overfitting on a single seed, it is possible to pass a
+            parameter ``rng`` as 'int' or 'np.random.RandomState' to this function.
+            If this parameter is not given, the default random state is used.
+
```
`shuffle`is missing

PhMueller replied:
`shuffle`is missing

PhMueller commented on hpobench/benchmarks/mo/lm_benchmark.py:
```diff
@@ -0,0 +1,368 @@
+        ----------
+        configuration
+        fidelity: Dict, None
+            epoch: int - Values: [1, 81]
```
We only allow 81. 

Maybe this is something we should discuss: 
It could be okay to query also the performance for different epoch steps

PhMueller replied:
We only allow 81. 

Maybe this is something we should discuss: 
It could be okay to query also the performance for different epoch steps

PhMueller commented on hpobench/benchmarks/mo/lm_benchmark.py:
```diff
@@ -0,0 +1,368 @@
+                    used fidelities in this evaluation
+        """
+
+        # The result dict should contain already all necessary information -> Just swap the function value from valid
```
outdated comment.

PhMueller replied:
outdated comment.

PhMueller commented on hpobench/benchmarks/mo/lm_benchmark.py:
```diff
@@ -0,0 +1,368 @@
+        assert fidelity['epoch'] == 81, 'Only test data for the 50. epoch is available. '
+        ts_start = time.time()
+
+        self.rng = rng_helper.get_rng(rng)
```
```suggestion
        self.rng = rng_helper.get_rng(self.rng, rng)
```

PhMueller replied:
```suggestion
        self.rng = rng_helper.get_rng(self.rng, rng)
```

PhMueller commented on hpobench/benchmarks/mo/lm_benchmark.py:
```diff
@@ -0,0 +1,368 @@
+                         }
+                }
+
+    __all__ = ["LanguageModelBenchmark"]
```
Add a new line at the end of file.

-> checkstyle error

PhMueller replied:
Add a new line at the end of file.

-> checkstyle error

PhMueller commented on hpobench/container/recipes/mo/Singularity.LanguageModelBenchmark:
```diff
@@ -0,0 +1,30 @@
+
+    cd /home \
+    && mkdir data && cd data \
+    && wget https://raw.githubusercontent.com/pytorch/examples/master/word_language_model/data/wikitext-2/train.txt \
```
no need to download in container. The dataloader should download the data automatically to the users host system (defined data dir in the hpobenchrc file)

PhMueller replied:
no need to download in container. The dataloader should download the data automatically to the users host system (defined data dir in the hpobenchrc file)

PhMueller commented on hpobench/container/recipes/mo/Singularity.LanguageModelBenchmark:
```diff
@@ -0,0 +1,30 @@
+    && wget https://raw.githubusercontent.com/pytorch/examples/master/word_language_model/data/wikitext-2/valid.txt \
+    && wget https://raw.githubusercontent.com/pytorch/examples/master/word_language_model/data/wikitext-2/test.txt \
+    && cd /home \
+    && git clone https://github.com/ayushi-3536/HPOBench.git \
```
please link to automl and master

PhMueller replied:
please link to automl and master

PhMueller commented on hpobench/container/recipes/mo/Singularity.LanguageModelBenchmark:
```diff
@@ -0,0 +1,30 @@
+    && cd /home \
+    && git clone https://github.com/ayushi-3536/HPOBench.git \
+    && cd HPOBench \
+    && git checkout fair_adult \
```
shoudl be `lm`

PhMueller replied:
shoudl be `lm`

PhMueller commented on hpobench/container/recipes/mo/Singularity.LanguageModelBenchmark:
```diff
@@ -0,0 +1,30 @@
+
+
+%runscript
+    python -s /home/HPOBench/hpobench/container/server_abstract_benchmark.py mo.adult_benchmark $@
```
Adult is still linked here.

PhMueller replied:
Adult is still linked here.

PhMueller commented on hpobench/util/data_manager.py:
```diff
@@ -849,6 +848,77 @@ def _load(self) -> Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray, np.ndar
+        self._download()
+        self.X_train, self.X_valid, self.X_test = self._load()
+        self.logger.info(f'LanguageModelDataManager: Data successfully loaded after {time() - t:.2f}')
+        print(self.X_train.shape)
```
Please remove print statements

PhMueller replied:
Please remove print statements

PhMueller reviewed (changes requested):
I left some commments. could you please have a look at them. Thanks

ayushi-3536 committed:
-minor cleanup

ayushi-3536 committed:
-minor cleanup

ayushi-3536 committed:
-minor cleanup

ayushi-3536 committed:
-minor cleanup

pfistfl committed:
Add YAHPO Benchmark (#142)

* Add yahpo_gym w help from phmueller

Co-authored-by: PhMueller <muller-phil@gmx.net>

PhMueller committed:
ADD Multi-Objective Nasbench201 (v0.0.6) (#152)

Update the Nasbench201 benchmark to support Multi-Objective queries.

If you want to use the *single objective* Nasbench201 benchmark, you can query the SO version of this benchmark. 
Although, we have not changed the benchmark logic, you can also use the container v0.0.5 in your experiments to reproduce results from the old version of this benchmark.

ayushi-3536 committed:
Benchmark: Fair Adult from MO-ASHA (#148)

We add the benchmark from the MO-ASHA paper by Schmucker et al.

It is a MO benchmark, training an MLP on the Adult data set.

ayushi-3536 committed:
Multi Objective CNN benchmark: Flowers and Fashion (#147)

Added mo cnn benchmarks from bag of baseline paper

We deviate from the original benchmark in two points: 
* we return as cost only the  training time instead of the total elapsed time
* we return as objective for minimization instead of `-100 * accuracy` now `1 - accuracy` to achieve better output scalings. 

Co-authored-by: ayushi-3536 <ayushi-3536@github.com>
Co-authored-by: Philipp Müller <muller-phil@gmx.net>

PhMueller commented on extra_requirements/lm_benchmark.json:
```diff
@@ -0,0 +1,6 @@
+{
+  "lm": [
+    "torch==1.3.0",
+    "tqdm"
```
add a reasonable version number (you can use the same as for the flower cnn benchmark)

PhMueller commented on hpobench/benchmarks/mo/lm_benchmark.py:
```diff
@@ -0,0 +1,383 @@
+        elapsed_time = ts_start - time.time()
+
+        return {'function_value': {'log_perplexity': log_perplexity,
+                                   'accuracy': val_acc.item(),
```
please change to `1 - val_acc` so that it is a min problem.

PhMueller commented on hpobench/benchmarks/mo/lm_benchmark.py:
```diff
@@ -0,0 +1,383 @@
+        elapsed_time = ts_start - time.time()
+
+        return {'function_value': {'log_perplexity': log_perplexity,
+                                   'accuracy': test_acc.item(),
```
see objective func

PhMueller commented on hpobench/container/benchmarks/mo/lm_benchmark.py:
```diff
@@ -0,0 +1,12 @@
+from hpobench.container.client_abstract_benchmark import AbstractBenchmarkClient
+
+
+class LanguageModelBenchmark(AbstractBenchmarkClient):
```
Use here `AbstractMoBenchmarkClient`

PhMueller commented on hpobench/container/benchmarks/mo/lm_benchmark.py:
```diff
@@ -0,0 +1,12 @@
+    def __init__(self, **kwargs):
+        kwargs['benchmark_name'] = kwargs.get('benchmark_name', 'LanguageModelBenchmark')
+        kwargs['container_name'] = kwargs.get('container_name', 'lm_benchmark')
+        kwargs['latest'] = kwargs.get('container_tag', '0.0.1')
```
add

`kwargs['gpu'] = kwargs.get('gpu', True)`

PhMueller commented on hpobench/dependencies/lm/model.py:
```diff
@@ -0,0 +1,148 @@
+class TransformerModel(nn.Module):
+    """Container module with an encoder, a transformer module, and a decoder."""
+
+    def __init__(self, ntoken, ninp, nhead, nhid, nlayers, dropout=0.5, bptt=35, rng=None):
```
signature if possible

PhMueller reviewed (changes requested):
it would be cool if you could go through the comments. Thanks

ayushi-3536 committed:
-add gpu support
-add dependency version

ayushi-3536 committed:
-add gpu support
-add dependency version

ayushi-3536 committed:
-add MO abstract client

ayushi-3536 committed:
resolve conflict

ayushi-3536 committed:
- minimize objective values
